LibraBFT/Impl/Consensus/BlockStorage/BlockStore.agda

```diff
@@ -19,11 +20,24 @@ open import Optics.All
 module LibraBFT.Impl.Consensus.BlockStorage.BlockStore where
 
 postulate
-  executeAndInsertBlockE : ∀ {𝓔} → BlockStore 𝓔 → Block → Either ErrLog (BlockStore 𝓔 × ExecutedBlock)
   insertTimeoutCertificateM : TimeoutCertificate → LBFT (Either ErrLog Unit)
-  getBlock : ∀ {𝓔 : EpochConfig} → HashValue → BlockStore 𝓔 → Maybe ExecutedBlock
   getQuorumCertForBlock : ∀ {𝓔 : EpochConfig} → HashValue → BlockStore 𝓔 → Maybe QuorumCert
 
+------------------------------------------------------------------------------
+
+getBlock : ∀ {𝓔 : EpochConfig} → HashValue → BlockStore 𝓔 → Maybe ExecutedBlock
+
+executeAndInsertBlockE
+  : ∀ {𝓔}
+  → BlockStore 𝓔 → Block
+  → Either ErrLog (BlockStore 𝓔 × ExecutedBlock)
+
+executeBlockE : ∀ {𝓔 : EpochConfig} → BlockStore 𝓔 → Block → Either ErrLog ExecutedBlock
+
+pathFromRoot : ∀ {𝓔 : EpochConfig} → HashValue → BlockStore 𝓔 → Either ErrLog (List ExecutedBlock)
```

Comment:
Curious why this returns `Either ErrLog (List ExecutedBlock)`, whereas the Haskell code returns `Maybe [ExecutedBlock]`.

Reply:
Your Haskell branch is out of date.

Reply:
Ah yes, you're right.  One commit behind :smile:.  Thanks.